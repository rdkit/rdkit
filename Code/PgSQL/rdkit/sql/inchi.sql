select mol_inchi('c1ccccc1'::mol);
select mol_inchikey('c1ccccc1'::mol);
select mol_inchi('Cc1cc(C)[n+]c(C)c1'::mol);
select mol_inchikey('Cc1cc(C)[n+]c(C)c1'::mol);
select mol_inchi(mol_from_ctab((Chr(10) || Chr(10) || Chr(10) ||
'  0  0  0  0  0  0  0  0  0  0999 V2000' ||
Chr(10) ||
'M  END')::cstring));
select mol_inchi('');
select mol_inchikey('');

