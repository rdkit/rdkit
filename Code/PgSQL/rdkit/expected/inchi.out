select mol_inchi('c1ccccc1'::mol);
             mol_inchi              
------------------------------------
 InChI=1S/C6H6/c1-2-4-6-5-3-1/h1-6H
(1 row)

select mol_inchikey('c1ccccc1'::mol);
        mol_inchikey         
-----------------------------
 UHOVQNZJYSORNB-UHFFFAOYSA-N
(1 row)

select mol_inchi('Cc1cc(C)[n+]c(C)c1'::mol);
                      mol_inchi                       
------------------------------------------------------
 InChI=1S/C8H11N/c1-6-4-7(2)9-8(3)5-6/h4-5H,1-3H3/q+1
(1 row)

select mol_inchikey('Cc1cc(C)[n+]c(C)c1'::mol);
        mol_inchikey         
-----------------------------
 WDQXTFQYHUZLPX-UHFFFAOYSA-N
(1 row)

select mol_inchi(mol_from_ctab((Chr(10) || Chr(10) || Chr(10) ||
'  0  0  0  0  0  0  0  0  0  0999 V2000' ||
Chr(10) ||
'M  END')::cstring));
 mol_inchi  
------------
 InChI=1S//
(1 row)

select mol_inchi('');
 mol_inchi  
------------
 InChI=1S//
(1 row)

select mol_inchikey('');
        mol_inchikey         
-----------------------------
 MOSFIJXAXDLOML-UHFFFAOYSA-N
(1 row)

select mol_inchi('CC1=NC=CN1'::mol);
                    mol_inchi                     
--------------------------------------------------
 InChI=1S/C4H6N2/c1-4-5-2-3-6-4/h2-3H,1H3,(H,5,6)
(1 row)

select mol_inchikey('CC1=NC=CN1'::mol);
        mol_inchikey         
-----------------------------
 LXBGSDVWAMZHDD-UHFFFAOYSA-N
(1 row)

select mol_inchi('CC1=NC=CN1'::mol,'/FixedH');
                       mol_inchi                       
-------------------------------------------------------
 InChI=1/C4H6N2/c1-4-5-2-3-6-4/h2-3H,1H3,(H,5,6)/f/h5H
(1 row)

select mol_inchikey('CC1=NC=CN1'::mol,'/FixedH');
        mol_inchikey         
-----------------------------
 LXBGSDVWAMZHDD-JSWHHWTPNA-N
(1 row)

